src/MongoDB.Driver/Linq/Translators/AggregateLanguageTranslator.cs

```diff
@@ -596,6 +611,7 @@ private BsonValue TranslateSelect(SelectExpression node)
                 return inputValue.ToString() + inValue.ToString().Substring(node.ItemName.Length + 2);
             }
 
+            _templatesState.AddMapping(node.ItemName);
```

Comment:
This saving is required to understand whether the out of scope parameter is defined in the top level or not. If there is mapping - the out of scope parameter was defined inside a not top-level expression